packages/webview_flutter/webview_flutter_web/test/webview_flutter_web_test.dart

```diff
@@ -54,17 +55,17 @@ void main() {
       verify(mockElement.src = 'test url');
     });
 
-    test('loadHtmlString loads html into iframe', () {
+    test('loadHtmlString escapes "#" correctly', () {
       // Setup
       final MockIFrameElement mockElement = MockIFrameElement();
       final WebWebViewPlatformController controller =
           WebWebViewPlatformController(
         mockElement,
       );
       // Run
-      controller.loadHtmlString('test html');
+      controller.loadHtmlString('#');
       // Verify
-      verify(mockElement.src = 'data:text/html,${Uri.encodeFull('test html')}');
+      verify(mockElement.src = 'data:text/html;charset=utf-8,%23');
     });
 
     group('loadRequest', () {
```

Comment:
And there should be a second version of the test that tests the changes you made to `loadRequest`.